v2/contrib/net/http/option.go

```diff
@@ -51,15 +73,15 @@ func defaults(cfg *config) {
 
 // WithIgnoreRequest holds the function to use for determining if the
 // incoming HTTP request should not be traced.
-func WithIgnoreRequest(f func(*http.Request) bool) Option {
-	return func(cfg *config) {
+func WithIgnoreRequest(f func(*http.Request) bool) OptionFn {
```

Comment:
I'm not very fond of having this *Fn types but I don't see a better way to improve the API signatures while ensuring early detection of any potential misuse.